pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -213,6 +215,14 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
     }
   }
 
+  public static boolean shouldDeriveNumDocsPerChunk(String columnName, Map<String, Map<String, String>> columnProperties) {
+    if (columnProperties != null) {
+      Map<String, String> properties = columnProperties.get(columnName);
+      return properties != null && Boolean.parseBoolean(properties.get(FieldConfig.DERIVE_NUM_DOCS_PER_CHUNK_RAW_INDEX_KEY));
```

Comment:
Defaults to false, right?

Reply:
Can we derive it automatically? (e.g. if column is text index then we derive it from metadata) Or, do you see this being usefiul in other cases as well?

Reply:
> Defaults to false, right?

Yes

> Can we derive it automatically? (e.g. if column is text index then we derive it from metadata) Or, do you see this being usefiul in other cases as well?

We could. Even for columns with text indexes, I don't think we should use it by default (since now that we have seen the potential -ve impact related to access pattern). Yes, most likely this will be used for columns with text index but only if the average column value size is very large (around 1-2MB) since that is the case which takes the chunk size and compressed chunk size (2 * raw) > 2GB and deriving the numDocsPerChunk becomes useful.

Reply:
@mcvsubbu deriving was the issue for our production issue.

Reply:
The reason I ask is that if we introduce a config it is hard to remove/deprecate, etc. if we make it a default for text column, we can always introduce a config later to adjust. In both offline and realtime cases, we know the average column size (or, can compute easily) at the segment generation time, so it seems to me that this can be done automatically without introducing a configuration. I would propose to NOT introduce a config at this time